esx_service/cli/vmdkops_admin_test.py

```diff
@@ -901,48 +955,52 @@ def test_tenant_access(self):
             expected_output = self.datastore_name
             self.assertEqual(expected_output, actual_output)
 
-            # add second access privilege for tenant
-            # allow_create = False
-            # max_volume size = 600MB
-            # total_volume size = 1GB
-            volume_maxsize_in_MB = convert.convert_to_MB("600MB")
-            volume_totalsize_in_MB = convert.convert_to_MB("1GB")
-            error_info = auth_api._tenant_access_add(name=self.tenant1_name,
-                                                    datastore=self.datastore1_name,
-                                                    default_datastore=True,
-                                                    allow_create=False,
-                                                    volume_maxsize_in_MB=volume_maxsize_in_MB,
-                                                    volume_totalsize_in_MB=volume_totalsize_in_MB
-                                                )
+            # update the "default_datastore"
+            error_info = auth_api._tenant_update(name=self.tenant1_name,
+                                                 default_datastore=self.datastore1_name)
             self.assertEqual(None, error_info)
 
             error_info, tenant_list = auth_api._tenant_ls()
             self.assertEqual(None, error_info)
 
-
             rows = vmdkops_admin.generate_tenant_ls_rows(tenant_list)
             # get "default_datastore" from the output
             actual_output = rows[1][3]
             expected_output = self.datastore1_name
             self.assertEqual(expected_output, actual_output)
 
+            # swtich the "default_datastore" to self.datastore_name
+            error_info = auth_api._tenant_update(name=self.tenant1_name,
+                                                 default_datastore=self.datastore_name)
+            self.assertEqual(error_info, None)
+            # remove the privilege to self.datastore1_name
             error_info = auth_api._tenant_access_rm(name=self.tenant1_name,
                                                     datastore=self.datastore1_name)
             self.assertEqual(error_info, None)
 
-        # remove access privileges
+        # remove access privileges, which should fail
+        # since the "default_datastore" is set to self.datastore_name
+        # cannot remove the privilege
         error_info = auth_api._tenant_access_rm(name=self.tenant1_name,
         datastore=self.datastore_name)
 
-        self.assertEqual(None, error_info)
+        self.assertNotEqual(None, error_info)
```

Comment:
Add a test where the default datastore is changed and tenant_access_rm()  is done on the earlier DS, that should work as its no longer the default DS.